player/configfiles.c

```diff
@@ -366,15 +364,15 @@ void mp_write_watch_later_conf(struct MPContext *mpctx)
     } else {
         fprintf(file, "start=%f\n", pos);
     }
-    for (int i = 0; backup_properties[i]; i++) {
-        const char *pname = backup_properties[i];
+    for (int i = 0; opts->watch_later_properties[i]; i++) {
+        const char *pname = opts->watch_later_properties[i];
```

Comment:
the key is that we shouldn't touch `opts->watch_later_properties` here. since the `mpctx->resume_defaults` has both the name and the default value, we can just iterate through `mpctx->resume_defaults`(add a sentinel element to make it null-terminated). so we are not affected by the the `opts->watch_later_properties` changes.

Reply:
But why would we want that if this is supposed to support runtime changes?

Reply:
I explained the meaning of runtime change support. If you want the true runtime change support, you should backup all properties first.

Reply:
I don't mind if it supports runtime changes or not, since I only need this in `mpv.conf` as I suspect most users do, and didn't mean to support runtime changes at first. Copying the existing resume defaults and calculating any new one when `watch-later-properties` is modified at runtime is what Dudemanguy suggested on IRC, and he said storing every option and property at startup is silly, so it's up to him or the other maintainers to decide which version of this to use.

Reply:
Storing every single option/property on startup could have a measurable impact on startup times which is definitely unacceptable for this feature imo. I am perfectly fine with the caveat explained in the manpage and it's easy for a user to workaround it.